bleak/backends/bluezdbus/manager.py

```diff
@@ -1035,22 +1029,16 @@ def _parse_msg(self, message: Message) -> None:
                                     message_path, new_value
                                 )
 
-    def _run_advertisement_callbacks(
-        self, device_path: str, device: Device1, changed: Iterable[str]
-    ) -> None:
+    def _run_advertisement_callbacks(self, device_path: str, device: Device1) -> None:
         """
         Runs any registered advertisement callbacks.
 
         Args:
             device_path: The D-Bus object path of the remote device.
             device: The current D-Bus properties of the device.
-            changed: A list of properties that have changed since the last call.
         """
-        for callback, adapter_path in self._advertisement_callbacks:
-            # filter messages from other adapters
-            if adapter_path != device["Adapter"]:
-                continue
-
+        adapter_path = device["Adapter"]
+        for callback in self._advertisement_callbacks[adapter_path]:
```

Comment:
Average [time complexity](https://wiki.python.org/moin/TimeComplexity) should now be: `O(callbacks)` instead of `O(adapters * callbacks)`

Reply:
Do you have before and after flamegraphs or something like that showing the difference?

Reply:
https://github.com/hbldh/bleak/pull/1632#issuecomment-2303402217